examples/big_bank.py

```diff
@@ -16,7 +16,8 @@
 """
 Provide a 'Big Bank' example.
 
-Illustrate how to create a software architecture diagram using code.
+Illustrate how to create a software architecture diagram using code, based on
+https://github.com/structurizr/dsl/blob/master/examples/big-bank-plc.dsl.
```

Comment:
So actually, we based the example on: https://github.com/structurizr/dotnet/blob/master/Structurizr.Examples/BigBankPlc.cs

And there, there's no mention of ids. I'm not event sure why we would need them if "name" is a unique identifier?

I don't know the DSL that well, but it seems that there's no real usage of the id, as the name is unique as well.

Can you elaborate more on this?

Reply:
Sure.  One of the things I'm working on is "architecture as code", including integrating architecture change into our other change management processes across the enterprise.  For a given system, I want to be able to have a workspae that combines together high-level structure (Systems, Containers) which would be either defined through the Python API or in Structurizr itself, along with lower-level (Component, Code) which would be extracted from source code directly through the [Component Finder](https://github.com/structurizr/java-extensions/blob/master/docs/component-finder.md), [annotations](https://github.com/structurizr/java-extensions/blob/master/docs/structurizr-annotations.md) and so on.

Ultimately I want to have a file which can be managed in Git (and so have PR approval workflows) that represents the whole model, so the two obvious choices are JSON and the Structurizr DSL.  The JSON form isn't really designed for humans, and if you look at a typical diff then it's pretty hard to tell what's going on, but with the DSL then it's clear what is changing.  As long as you have IDs on things, else the Relationships become pretty opaque.  

Here's an example from BigBank, using a rudimentary DSL translator that I knocked together:
```
      1 -> 2 Uses
      2 -> 4 Uses
      2 -> 6 "Sends e-mail using"
      6 -> 1 "Sends e-mails to"
      9 -> 4 Uses
      1 -> 9 "Withdraws cash using"
```
and here's what it looks like if we support IDs:
```
      customer -> internetBankingSystem "Views account balances, and makes payments using"
      internetBankingSystem -> mainframe "Gets account information from, and makes payments using"
      internetBankingSystem -> email "Sends e-mail using"
      email -> customer "Sends e-mails to"
      atm -> mainframe Uses
      customer -> atm "Withdraws cash using"
```
That second version is way more understandable in a diff than the first, which is why I'd like to support it.  I'm surprised that the .cs version isn't consistent with the DSL to be honest!